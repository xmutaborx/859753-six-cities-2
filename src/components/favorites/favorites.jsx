import React from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header.jsx';
import FavoritesEmpty from '../favorites-empty/favorites-empty.jsx';
import OffersList from '../offers-list/offers-list.jsx';
import {connect} from 'react-redux';
import Operations from '../../store/operations';
import {getFavoritesCitiesList} from '../../store/selectors';

class Favorites extends React.PureComponent {
  componentDidMount() {
    this.props.getFavorites();
  }

  render() {
    if (!this.props.favorites.length) {
      return <FavoritesEmpty />;
    }

    return (
      <div className="page">
        <Header />
        <main className="page__main page__main--favorites">
          <div className="page__favorites-container container">
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <OffersList
                favoritesMode={true}
                offers={this.props.favorites}
                favoritesCity={this.props.favoritesCity}
              />
            </section>
          </div>
        </main>
      </div>
    );
  }
}

Favorites.propTypes = {
  getFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.arrayOf(PropTypes.object),
  favoritesCity: PropTypes.arrayOf(PropTypes.string),
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  favorites: state.favorites,
  favoritesCity: getFavoritesCitiesList(state)
});

const mapDispatchToProps = (dispatch) => ({
  getFavorites: () => dispatch(Operations.getFavorites()),
});

export {Favorites};
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
