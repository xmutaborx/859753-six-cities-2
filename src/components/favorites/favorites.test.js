import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";

import {Favorites} from './favorites.jsx';
import OFFER_MOCK from '../../mocks/offer-mock';

Enzyme.configure({adapter: new Adapter()});

it(`Favorites correctly renders`, () => {
  const tree = shallow(<Favorites
    favorites={OFFER_MOCK}
    favoritesCity={[`Amsterdam`]}
    getFavorites={() => {}}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
