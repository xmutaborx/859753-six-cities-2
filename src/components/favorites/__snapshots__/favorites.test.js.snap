// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites correctly renders 1`] = `
<div
  className="page"
>
  <Connect(Header) />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <Connect(WithActiveItem)
          favoritesCity={
            Array [
              "Amsterdam",
            ]
          }
          favoritesMode={true}
          offers={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
    </div>
  </main>
</div>
`;
